src/views/Stack/Card.tsx

```diff
@@ -913,7 +913,9 @@ export default class Card extends React.Component<Props> {
                     contentStyle,
                   ]}
                 >
-                  <StackAnimationProgressContext.Provider value={current}>
+                  <StackAnimationProgressContext.Provider
+                    value={next || current}
```

Comment:
I think it's confusing to expose the progress of next screen or current screen interchangeably. We should be explicit about that.

Probably it'll be better to expose the same values that we expose to `cardStyleInterpolator`:

```tsx
type StackCardInterpolationProps = {
  current: { progress: Animated.Node<number>; };
  next?: { progress: Animated.Node<number>; };
  index: number;
  closing: Animated.Node<0 | 1>;
  layouts: {
    screen: Layout;
  };
  insets: {
    top: number;
    right: number;
    bottom: number;
    left: number;
  };
}
```

We can use a single `StackCardAnimationContext` and expose this object there. The object needs to be memoized to avoid unnecessary context updates similar to how `cardStyleInterpolator` is memoized.

Reply:
Yes, I agree - that would be really great. Should this replace the other contexts as well? (`StackAnimationIsSwipingContext` and `StackAnimationIsClosingContext`)?

Reply:
yea, we can remove them

Reply:
Something like this?:


```js
<StackCardAnimationContext.Provider value={{
    current: { progress: current },
    next: next ? { progress: next } : undefined,
    closing: this.isClosing,
    swiping: this.isSwiping,
    index: index,
    insets: this.props.insets,
    layouts: {
      screen: this.props.layout,
    },
  }}
>
  {children}
</StackCardAnimationContext.Provider>
```

Reply:
Yes. We also need to memoize this object to avoid updating this context on each render (like we memoize `cardStyleInterpolator`)

Reply:
Why should we memoize this? It is a really simple object that contains no extra calculations other than the values we already have - memoization might event add unnecessary overhead here.

Reply:
It's not about the object being expensive. Unless this is memoized, all components which use this context will re-render even if they don't need to when the `Card` component re-renders.

The components maybe expensive to re-render. If we don't memoize it here, it won't be possible to prevent the re-renders without changing code in those components.

We already have utility for memoization. Given that this object is only used once, it should be pretty simple to memoize.

Reply:
Would this work? `getCardAnimationContext` uses the same memoized pattern as styleInterpolator:

```js
 <StackCardAnimationContext.Provider
                    value={this.getCardAnimationContext(
                      index,
                      current,
                      next,
                      this.isClosing,
                      this.isSwiping,
                      this.props.layout,
                      this.props.insets
                    )}
                  >
                    {children}
                  </StackCardAnimationContext.Provider>v
```

Reply:
If this is unclear I can push this change.

Reply:
lgtm